src/VehicleSetup/FirmwareUpgradeController.cc

```diff
@@ -65,17 +80,25 @@ void FirmwareUpgradeController::startBoardSearch(void)
     _threadController->startFindBoardLoop();
 }
 
-void FirmwareUpgradeController::flash(FirmwareType_t firmwareType)
+void FirmwareUpgradeController::flash(AutoPilotStackType_t stackType,
+                                      FirmwareType_t firmwareType,
+                                      FirmwareVehicleType_t vehicleType)
 {
```

Comment:
When you get to here flashing a Flow, vehicleType is set to QuadFirmware instead of DefaultVehicleFirmware which in turn causes it to not find the firmware file. This was the same problem you fixed with Radio.